File: backend/src/main/java/com/gxs/bank/model/Beneficiary.java
package com.gxs.bank.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;
import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "beneficiaries")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Beneficiary {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    @JsonIgnore
    private User user;

    @Column(name = "beneficiary_name", nullable = false)
    private String beneficiaryName;

    @Column(name = "account_number", nullable = false)
    private String accountNumber;

    @Column(name = "ifsc_code", nullable = false)
    private String ifscCode;

    @Column(name = "bank_name")
    private String bankName;

    @Column(name = "nickname")
    private String nickname;

    @Column(name = "is_verified")
    private Boolean isVerified;

    @Column(name = "max_transfer_limit", precision = 15, scale = 2)
    private BigDecimal maxTransferLimit;

    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        if (isVerified == null) isVerified = false;
        if (maxTransferLimit == null) maxTransferLimit = new BigDecimal("100000.00");
    }
}

File: backend-python/com/gxs/bank/model/Beneficiary.py
from datetime import datetime
from decimal import Decimal
import uuid

from sqlalchemy import Boolean, Column, DateTime, ForeignKey, Numeric, String
from sqlalchemy.orm import relationship

from com.gxs.bank.config.database import Base


class Beneficiary(Base):
    __tablename__ = "beneficiaries"

    id = Column(String(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    userId = Column("user_id", String(36), ForeignKey("users.id"), nullable=False)
    user = relationship("User", lazy="select")

    beneficiaryName = Column("beneficiary_name", String(255), nullable=False)
    accountNumber = Column("account_number", String(255), nullable=False)
    ifscCode = Column("ifsc_code", String(255), nullable=False)
    bankName = Column("bank_name", String(255), nullable=True)
    nickname = Column("nickname", String(255), nullable=True)
    isVerified = Column("is_verified", Boolean, nullable=False, default=False)
    maxTransferLimit = Column(
        "max_transfer_limit",
        Numeric(15, 2),
        nullable=False,
        default=Decimal("100000.00"),
    )
    createdAt = Column("created_at", DateTime, nullable=False, default=datetime.utcnow)
